### ISSUE TITLE
Optimistic UI update issue

### ISSUE THREAD

**Issue by copleykj** [cult-of-coders/redis-oplog#202]
Client side insert operations cause a flicker in the UI as documents are added, removed and then finally added again.

**Comment by theodorDiaconu** [comment]
@copleykj can you please try Collection.insert(xxx, {optimistic: true})

**Comment by theodorDiaconu** [comment]
@copleykj also, is redis-oplog your top package ?

**Comment by theodorDiaconu** [comment]
Trying to run your socialize-demo:
```

   While selecting package versions:
   error: unknown package in top-level dependencies: socialize:requestable
   unknown package in top-level dependencies: socialize:user-blocking
   unknown package in top-level dependencies: socialize:postable
```

**Comment by theodorDiaconu** [comment]
Tried to delete them then:
```
While selecting package versions:
   error: Potentially incompatible change required to top-level dependency: socialize:friendships 0.4.3, was 1.0.0.
   Constraints on package "socialize:friendships":
   
   Potentially incompatible change required to top-level dependency: socialize:messaging 0.5.4, was 1.0.0.
   Constraints on package "socialize:messaging":
   
   Potentially incompatible change required to top-level dependency: socialize:feed 0.2.3, was 1.0.0.
   Constraints on package "socialize:feed":
   
   Potentially incompatible change required to top-level dependency: socialize:user-model 0.1.7, was 1.0.0.
   Constraints on package "socialize:user-model":
   * socialize:user-model@0.1.7 <- socialize:friendships 0.4.3
   * socialize:user-model@0.1.7 <- socialize:messaging 0.5.4
   * socialize:user-model@0.1.7 <- socialize:user-presence 0.4.0
   * socialize:user-model@0.1.7 <- socialize:feed 0.2.3
   
   Potentially incompatible change required to top-level dependency: socialize:likeable 0.2.1, was 1.0.0.
   Constraints on package "socialize:likeable":
   * socialize:likeable@0.2.1 <- socialize:commentable 0.2.1
   
   Potentially incompatible change required to top-level dependency: socialize:commentable 0.2.1, was 1.0.0.
   Constraints on package "socialize:commentable":
   * socialize:commentable@0.2.1 <- socialize:feed 0.2.3
   
   Potentially incompatible change required to top-level dependency: socialize:user-presence 0.4.0, was 1.0.0.
   Constraints on package "socialize:user-presence":
   * socialize:user-presence@0.3.4 <- socialize:messaging 0.5.4
```

**Comment by theodorDiaconu** [comment]
Another important question, how are you subscribing to the data, is it by any chance a publishComposite ? :D

**Comment by copleykj** [comment]
Yes redis-oplog is top package. It seems this happens when insert has a channel option but the publication isn't using the channel. Oddly enough adding the optimistic option fixes the issue..

as far as the error, the demo currently requires packages that aren't available on atmosphere so there is a script that clones the necessary packages from github to the packages folder.

**Comment by copleykj** [comment]
no, not yet using publish-composite.. that's my next step

**Comment by copleykj** [comment]
I'm gonna reopen this as it's become an issue again once I switched every over to the new collection level configuration.

**Comment by theodorDiaconu** [comment]
@copleykj indeed, this is fixed now in feature/1.2.6 cheers

**Comment by copleykj** [comment]
Yes! I had checked out the release/1.2.6 branch this morning and it works fantastically. Thanks so much for all your hard work.:blush:
